src/RefractDataStructure.cc

```diff
@@ -853,6 +874,49 @@ namespace
         element.attributes().set(key, fetch(std::move(values.back())));
     }
 
+    void CheckForMultipleDefaultDefinitions(
+        const ElementInfoContainer<refract::EnumElement>& values, ConversionContext& context)
+    {
+        if (values.empty()) {
+            return;
+        }
+
+        if ((values.size() > 1) || (values.front().value.size() > 1)) {
+
+            mdp::CharactersRangeSet location;
+            for (const auto& item : values) {
+                location.append(item.sourceMap.sourceMap);
+            }
+
+            context.warn(snowcrash::Warning("multiple definitions of 'default' value", snowcrash::MSONError, location));
+        }
+    }
+
+    template <>
+    void LastElementToAttribute<refract::EnumElement, ElementInfoContainer<refract::EnumElement> >(
+        ElementInfoContainer<refract::EnumElement> values,
+        const std::string& key,
+        IElement& element,
+        ConversionContext& context)
+    {
+        using T = refract::EnumElement;
+
+        CheckForMultipleDefaultDefinitions(values, context);
+
+        if (values.empty()) {
+            return;
+        }
+
+        auto info = Merge<T>()(std::move(values));
+
+        if (info.value.empty()) {
+            element.attributes().set(key, make_empty<EnumElement>());
+            return;
+        }
+
+        element.attributes().set(key, make_element<EnumElement>(dsd::Enum{ std::move(info.value.back()) }));
```

Comment:
Should we pop the last element after the move? Its value is `nullptr`.

Reply:
We can, but in this case rest is dropped, so we can omit pop in this case
It is same like nonspecialized case. see:
https://github.com/apiaryio/drafter/blob/ce0edbd63e9ef3e8d9249929d7031666cba74b24/src/RefractDataStructure.cc#L874